File: rust/src/bin/p0003.rs
// Project Euler: Problem 3

euler::run_solver!(3);

fn solve() -> String {
    compute(600_851_475_143).to_string()
}

fn compute(n: i64) -> i64 {
    use euler::math;

    debug_assert!(n > 0);

    *math::prime_factors(n).last().unwrap()
}

#[cfg(test)]
mod tests {
    use super::compute;

    #[test]
    fn p0003_13195() {
        assert_eq!(compute(13_195), 29);
    }

    #[test]
    fn p0003_600851475143() {
        assert_eq!(compute(600_851_475_143), 6_857);
    }
}

File: ocaml/src/p0003.ml
(* Project Euler: Problem 3 *)

open Core

let compute num =
  let base, _ = Euler.Math.factorize num |> List.last_exn in
  base
;;

let solve () = compute 600851475143 |> Int.to_string

(* Test *)

let%test_unit "13195" = [%test_eq: int] (compute 13195) 29
let%test_unit "600851475143" = [%test_eq: int] (compute 600851475143) 6857
